cmd/spiffe-helper/config/config.go

```diff
@@ -105,8 +105,19 @@ func (c *Config) ValidateConfig() error {
 	}
 
 	if c.AgentAddress == "" {
-		c.AgentAddress = os.Getenv("SPIRE_AGENT_ADDRESS")
-		if c.AgentAddress == "" {
+		spireAgentAddress := os.Getenv("SPIRE_AGENT_ADDRESS")
+		spiffeEndpointSocket := os.Getenv("SPIFFE_ENDPOINT_SOCKET")
+
+		switch {
+		case spireAgentAddress != "" && spiffeEndpointSocket == "":
+			log.Warn("SPIRE_AGENT_ADDRESS is deprecated and will be removed in 0.10.0. Use SPIFFE_ENDPOINT_SOCKET instead.")
+			c.AgentAddress = spireAgentAddress
+		case spireAgentAddress != "" && spiffeEndpointSocket != "":
+			log.Warn("SPIRE_AGENT_ADDRESS and SPIFFE_ENDPOINT_SOCKET both set. Using value from SPIFFE_ENDPOINT_SOCKET. Support for SPIRE_AGENT_ADDRESS is deprecated and will be removed in 0.10.0.")
```

Comment:
can you return error here?
when both are set, how can we define witch one to use?

Reply:
Done